swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items_matching_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                items_matching_categories.append(item)
+        return items_matching_categories
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.append(my_item)
+            friend.inventory.remove(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if self.inventory and friend.inventory:
+            if self.swap_items(friend, self.inventory[0], friend.inventory[0]):
+                return True
+        else:
+            return False
+        
+    def get_best_by_category(self, category):
+        items_in_category = [item for item in self.inventory if item.category == category]
+        if len(items_in_category) == 0:
+            return None
+        
+        best_condition_item = items_in_category[0]
+        for item in items_in_category:
+            if item.condition > best_condition_item.condition:
+                best_condition_item = item
+
+        return best_condition_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        if self.get_best_by_category(their_priority) and other.get_best_by_category(my_priority):
+            self.swap_items(other, self.get_best_by_category(their_priority), other.get_best_by_category(my_priority))
+            return True
+        else:
+            return False
```

Comment:
Nice job reusing `get_best_by_category` and `swap_items`.

Here, I'd suggest storing the results of the first calls to `get_best_by_category` for use later on. Calling them twice isn't free, plus we might make a mistake in the way we call each the second time. So something like this (note also the use of a guard clause).

```python
    def swap_best_by_category(self, other, my_priority, their_priority):
        my_item = self.get_best_by_category(their_priority)
        their_item = other.get_best_by_category(my_priority)

        if not my_item or not their_item:
            return False

        return self.swap_items(other, my_item, their_item)
```

We could even observe that the behavior of `swap_best_by_category` is the same as `swap_items` (in how it returns `True` and `False`). Would that allow us to simplify the code even further? What would `swap_items` do if either input happened to be `None`?